### PR TITLE
Update npm naming

### PR DESCRIPTION
**Pull request by gabritto**
Updates to npm naming rule to be compatible with new dts-critic checks.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
   },
   "dependencies": {
     "definitelytyped-header-parser": "3.8.2",
-    "dts-critic": "^2.2.4",
+    "dts-critic": "file:../dts-critic",
```
Comment (gabritto) [inline package.json]: I'll have to update this once we publish a new version of dts-critic to npm.

File: src/rules/npmNamingRule.ts [hunk 2]
```diff
@@ -1,25 +1,177 @@
...
+                            enum: [Mode.Code],
+                        },
+                        "errors": {
+                            type: "array",
+                            items: {
+                                type: "array",
+                                items: [
+                                    {   description: "Name of the check.",
+                                        type: "string",
+                                        enum: [ErrorKind.NeedsExportEquals, ErrorKind.NoDefaultExport] as ExportErrorKind[],
```
Comment (sandersn) [inline src/rules/npmNamingRule.ts]: are these the only two error kinds that the lint rule should support?

File: src/rules/npmNamingRule.ts [hunk 3]
```diff
@@ -1,25 +1,177 @@
...
+    if (!arg.mode || typeof arg.mode !== "string") {
+        return defaultOptions;
+    }
+
+    const mode = parseMode(arg.mode);
+    if (!mode) {
+        return defaultOptions;
+    }
+
+    let singleLine;
```
Comment (sandersn) [inline src/rules/npmNamingRule.ts]:
```suggestion
    const singleLine = arg["single-line"] ? true : undefined;
```

or maybe just `!!arg["single-line"]`, I don't know if the `undefined` is ever important.

File: src/rules/npmNamingRule.ts (hunk 2 above)
Comment (gabritto) [reply src/rules/npmNamingRule.ts]: Yes, at least for now.

**Review by haaami01 — COMMENTED** [review]
Update your documents. I hope you're meeting your expectations exactly.

**Pull request merged**